contrib/mixin/manifests/etcd-prometheusRules.yaml

```diff
@@ -0,0 +1,147 @@
+groups:
```

Comment:
Please don't add generated files to repo

Reply:
That was the main intent, have it here so we can use it in https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/hack/sync_prometheus_rules.py#L63

Reply:
Then we should add a test that ensures that this file is up to date, so that PRs that change rules always need to regenerate this file.

I would like a second opinion of fact that we are adding this file. @tomwilkie @lilic

Reply:
I agree with @serathius here, you should rarely want to use a generic generated yaml and often override configs like namespaces and other fields. and @mrueg it might be better to sync on the non-generated files rather than yaml?

Reply:
In general I agree with not having rendered output in a repository (as it might run out of sync), this approach follows what https://github.com/prometheus-operator/kube-prometheus/tree/main/manifests and https://github.com/monitoring-mixins/website have been doing and had been practice in the etcd-3.4 and previous releases https://raw.githubusercontent.com/etcd-io/website/master/content/en/docs/v3.4/op-guide/etcd3_alert.rules.yml

Reply:
Yes, we still have [that issue in kube-prometheus](https://github.com/prometheus-operator/kube-prometheus/issues/509) and we are looking for alternatives to remove generated files from the repository. However, this is not a top priority right now.
As for mixins, https://monitoring.mixins.dev/ is regenerating etcd mixin daily and allows for consumption in YAML format (YAML file available in [here](https://github.com/monitoring-mixins/website/blob/master/assets/etcd/alerts.yaml))

Reply:
As for using it in https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/hack, why not add jsonnet execution there and customize mixin to the needs of the helm chart?